### PR TITLE
Introduction of inclusion mode for new devices

### PR DESCRIPTION
**Pull request by marcelrv**
S Command for setting MAX! thermostat Temperature Settings.
C Command to request config update
A Command to reset config
Test scripts
Fix formatting
Cube decoding

Signed-off-by: Marcel Verpaalen <marcel@verpaalen.com>

### CODE REVIEW

File: addons/binding/org.openhab.binding.max/ESH-INF/thing/bridge.xml [hunk 1]
```diff
@@ -86,8 +87,23 @@
 				<default>1000</default>
 				<advanced>true</advanced>
 			</parameter>
+
+			<!-- Trigger inclusion mode action from habmin menu -->
+			<parameter name="action-cubeInclusion" type="integer" groupName="actions">
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.max/ESH-INF/thing/bridge.xml]: Wouldn't it be much better to have inclusion done through discovery?
I.e. background discovery simply reads what is available on the cube, but when actively triggering a discovery, the cube should go in inclusion mode. I think this is how it is done for ZWave as well and also [the hue reference implementation](https://github.com/eclipse/smarthome/blob/master/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java#L78).

Comment (marcelrv) [inline addons/binding/org.openhab.binding.max/ESH-INF/thing/bridge.xml]: That makes sense. I'll update the PR
Op 13 mrt. 2016 11:38 schreef "Kai Kreuzer" <notifications@github.com>:

**Comment by marcelrv** [comment]
@kaikreuzer updated to use discovery to start inclusion mode

**Comment by marcelrv** [comment]
this PR includes further development for enhancement #201

**Comment by marcelrv** [comment]
PR extended with several other developments
S Command for setting MAX! thermostat Temperature Settings with tests …
C Command to request config update
A Command to reset config
T Command to delete device with test
Z Command to wakeup device with test
M Command to set room and device naming with
Q Command to quit connection
sendcommands improvement to allow devices to send all commands to cube
Fix formatting
Cube decoding

**Comment by marcelrv** [comment]
Seems like there is a jenkins issue unrelated to this PR

[INFO] MAX! Binding ....................................... SUCCESS [  2.806 s]
[INFO] MAX! Binding Tests ................................. SUCCESS [ 17.715 s]

but...
```
[INFO] MAX! Binding ....................................... SUCCESS [  2.806 s]
[INFO] MAX! Binding Tests ................................. SUCCESS [ 17.715 s]
[INFO] Network Binding .................................... SUCCESS [  3.711 s]
[INFO] OpenSprinkler Binding .............................. SUCCESS [  4.046 s]
[INFO] PioneerAvr Binding ................................. SUCCESS [  1.693 s]
[INFO] Pulseaudio Binding ................................. SUCCESS [  1.649 s]
[INFO] RME Binding ........................................ SUCCESS [  0.490 s]
[INFO] Rfxcom Binding ..................................... SUCCESS [  2.905 s]
[INFO] SamsungTV Binding .................................. SUCCESS [  0.968 s]
[INFO] SMA EnergyMeter Binding ............................ SUCCESS [  0.506 s]
[INFO] SqueezeBox Binding ................................. SUCCESS [  1.897 s]
[INFO] Tesla Binding ...................................... SUCCESS [  2.701 s]
[INFO] Vitotronic Binding ................................. SUCCESS [  0.525 s]
[INFO] ZWave Binding ...................................... SUCCESS [ 11.820 s]
[INFO] openHAB I/O Add-Ons ................................ SUCCESS [  0.049 s]
[INFO] my.openHAB Connection Service ...................... SUCCESS [  0.558 s]
[INFO] Hue Emulation Service .............................. SUCCESS [  0.792 s]
[INFO] HomeKit Integration ................................ SUCCESS [  6.591 s]
[INFO] openHAB UI Components .............................. SUCCESS [  0.047 s]
[INFO] openHAB Classic UI Dashboard Integration ........... SUCCESS [  1.166 s]
[INFO] openHAB Basic UI Dashboard Integration ............. SUCCESS [  0.484 s]
[INFO] openHAB Paper UI Theme Fragment .................... SUCCESS [  0.510 s]
[INFO] openHAB CometVisu Backend .......................... SUCCESS [  8.832 s]
[INFO] PHP Support for the CometVisu Backend .............. SUCCESS [  1.842 s]
[INFO] openHAB Features ................................... SUCCESS [  0.044 s]
[INFO] openHAB Feature Addons External .................... SUCCESS [  0.302 s]
[INFO] openHAB 2 Feature Addons ........................... SUCCESS [  2.763 s]
[INFO] openHAB 2 Feature Verification ..................... FAILURE [  6.266 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 08:59 min
[INFO] Finished at: 2016-03-20T15:38:03+01:00
[INFO] Final Memory: 46M/153M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project openhab2-addons-verify: Could not resolve dependencies for project org.openhab.addons:openhab2-addons-verify:pom:2.0.0-SNAPSHOT: Could not find artifact org.openhab.core:openhab-core:xml:features:2.0.0.x in cloudbees-private-release-repository (https://repository-openhab.forge.cloudbees.com/release) -> [Help 1]
[ERROR]
```

**Comment by kaikreuzer** [comment]
Please have a try with https://github.com/openhab/openhab2-addons/pull/751

**Comment by marcelrv** [comment]
@kaikreuzer yes, indeed i rebased and now fine again. thx

**Pull request merged**